src/XMakeBuildEngine/Definition/Project.cs

```diff
@@ -2162,6 +2176,14 @@ internal bool SplitItemElementIfNecessary(ProjectItemElement itemElement)
             return true;
         }
 
+        internal bool ItemElementRequiresSplitting(ProjectItemElement itemElement)
+        {
+            var hasCharactersThatRequireSplitting = FileMatcher.HasWildcardsSemicolonItemOrPropertyReferences(itemElement.Include);
+            var hasMoreThanOneItem = Items.Count(i => i.Xml == itemElement) > 1;
```

Comment:
Ok, if it's not super clear why this works, at least the variable name declares a clear intention.